controller/search_blackbox_test.go

```diff
@@ -942,3 +942,75 @@ func (s *searchBlackBoxTest) TestIncludedParents() {
 	}
 	assert.Equal(s.T(), successCnt, 4)
 }
+
+func (s *searchBlackBoxTest) TestFilterAssigneeNullAfterWIUpdate() {
+	resetFn := s.DisableGormCallbacks()
+	defer resetFn()
+
+	fxt := tf.NewTestFixture(s.T(), s.DB, tf.CreateWorkItemEnvironment(), tf.WorkItems(1))
+	spaceInstance := fxt.Spaces[0]
+	spaceIDStr := spaceInstance.ID.String()
+	filter := fmt.Sprintf(`
+							{"assignee":null}`,
+	)
+	_, result := test.ShowSearchOK(s.T(), nil, nil, s.controller, &filter, nil, nil, nil, nil, &spaceIDStr)
+	compareWithGoldenUUIDAgnostic(s.T(), filepath.Join(s.testDir, "show", "filter_assignee_null_create_work_item.golden.json"), result)
+	wi := result.Data[0]
+	workitemCtrl := NewWorkitemController(s.svc, gormapplication.NewGormDB(s.DB), s.Configuration)
+
+	wi.Attributes[workitem.SystemTitle] = "Updated Test WI"
+	updatedDescription := "= Updated Test WI description"
+	wi.Attributes[workitem.SystemDescription] = updatedDescription
+	payload2 := app.UpdateWorkitemPayload{
+		Data: &app.WorkItem{
+			Type:       APIStringTypeWorkItem,
+			Attributes: map[string]interface{}{},
+			Relationships: &app.WorkItemRelationships{
+				Space: app.NewSpaceRelation(spaceInstance.ID, ""),
+			},
+		},
+	}
+	payload2.Data.ID = wi.ID
+	payload2.Data.Attributes = wi.Attributes
+	_, updated := test.UpdateWorkitemOK(s.T(), s.svc.Context, s.svc, workitemCtrl, *wi.ID, &payload2)
+	compareWithGoldenUUIDAgnostic(s.T(), filepath.Join(s.testDir, "show", "filter_assignee_null_update_work_item.golden.json"), updated)
+
+	_, result = test.ShowSearchOK(s.T(), nil, nil, s.controller, &filter, nil, nil, nil, nil, &spaceIDStr)
+	compareWithGoldenUUIDAgnostic(s.T(), filepath.Join(s.testDir, "show", "filter_assignee_null_show_after_update_work_item.golden.json"), updated)
+}
+
+func (s *searchBlackBoxTest) TestFilterLabelNullAfterWIUpdate() {
+	resetFn := s.DisableGormCallbacks()
+	defer resetFn()
+
+	fxt := tf.NewTestFixture(s.T(), s.DB, tf.CreateWorkItemEnvironment(), tf.WorkItems(1))
+	spaceInstance := fxt.Spaces[0]
+	spaceIDStr := spaceInstance.ID.String()
+	filter := fmt.Sprintf(`
+							{"label":null}`,
+	)
+	_, result := test.ShowSearchOK(s.T(), nil, nil, s.controller, &filter, nil, nil, nil, nil, &spaceIDStr)
+	compareWithGoldenUUIDAgnostic(s.T(), filepath.Join(s.testDir, "show", "filter_label_null_create_work_item.golden.json"), result)
```

Comment:
why do you compare the result here ?

Reply:
Well, this is not really the code under test.  Earlier I used some `require` (`FailNow`) here, but found golden files more convenient. So, should I remove this?

Reply:
Removed it here. 735a0e884d9b98c42055953f1a2545d50d1aa86b